src/app/excel/excel-table/excel-table.component.ts

```diff
@@ -61,44 +68,44 @@ export class ExcelTableComponent implements OnInit {
       this.router.navigate([ 'projects' ]);
       return;
     }
-    this.sprintsService.getSprints(this.currentProject.id, [ 'DECLARABLE', 'PADDING', 'IN_PROGRESS', 'FINISHED' ])
+    this.sprintsService.getSprints(this.currentProject.id,
+      ['DECLARABLE', 'PADDING', 'IN_PROGRESS', 'FINISHED', 'UPCOMING', 'CLOSED'],
+      'ASC')
       .subscribe(sprints => {
         this.isLoaded = true;
         if (sprints.length === 0) {
           return;
         }
-        this.declarableSprint = sprints[0];
-        this.sprintStartDate = this.declarableSprint.durationPeriod.start;
-        this.sprintEndDate = this.declarableSprint.durationPeriod.end;
-
-        this.excelService.getSprintSummary(this.currentProject.id, this.declarableSprint.id)
-          .subscribe(summary => {
-            this.hoursSum = Math.floor(summary.totalDeclaredTime / 60) + Math.round((summary.totalDeclaredTime % 60) / 60 * 100) / 100;
-            this.actualRate = summary.prevAverageSprintCoefficient;
-
-            this.projectService.getMembers(this.currentProject.id).subscribe(members => {
-              summary.membersAvailability
-                .map((a: UserAvailabilityDto) => {
-                  const user = members.find((m: ProjectMemberDto) => m.user.id === a.userId);
-                  const worker: Worker = a.availability ?
-                    {
-                      timeAvailable: this.formatNumber(a.availability.timeAvailable),
-                      timeRemaining: this.formatNumber(a.availability.timeRemaining),
-                      notes: a.availability.notes,
-                      name: this.createFullName(user),
-                    } :
-                    { timeAvailable: '', timeRemaining: '', notes: '', name: this.createFullName(user) };
-
-                  this.workers.push(worker);
-                });
-            });
-          });
+        this.allSprints = sprints;
+        this.currentSprintInd = 0;
+        this.allSprints.forEach((v, i) => {
```

Comment:
Po co `forEach`? Który z sprintów `DECLARABLE` wybierzesz?
Zapytanie wysyłasz o posortowaną listę sprintów, to może wystarczyłoby wziąć pierwszy jej element.

Reply:
Właśnie nie, robię to po to, żeby pierwsze zawsze wyświetlił się sprint, który teraz może deklarować (nie mogą być dwa `DECLARABLE` w tym samym czasie). Bo jak przychodzą mi wszystkie sprinty, to one przychodzą od najstarszego, a chcemy, żeby było przewinięte już do aktualnego

Reply:
A aktualny nie wiedziałem jak do końca wybrać, więc wybrałem w ten sposób, bo było najłatwiej

Reply:
Faktycznie, teraz patrzę jeszcze raz na twoją uwagę i powinna tu być inna funkcja, nie forEach

Reply:
Okej. Choć mam przeczucie, że dałoby się jakoś prościej :P

Reply:
W sumie mógłbym się odpytywać o każdy sprint osobno, ale nie wiem czy to jest prościej (szczególnie z mojej perspektywy)

Reply:
Aaaa, czyli tu zawsze pobierasz całość sprintów? Te prevy i nexty się nie dzieją?

Reply:
Faktycznie, aktualnie się nie dzieją, ale nie trzeba dużo chyba, żeby to zmienić

Reply:
Ja myślałam, żeby zapytać o jeden IN_PROGRESS, PADDING, DECLARABLE, FINISHED - dostałbyś pierwszy z nich i to pewnie je chcemy wyświetlać. Jakby nie było, to pierwszy UPCOMING. Jakby nie było, to ostatni CLOSED

Reply:
A no właśnie, już wiem czemu zrobiłem tak jak zrobiłem, no bo nie wiemy, w jakich stanach dla tego projektu są sprinty i właśnie, żeby nie odpytywać się o każdy stan osobno

Reply:
A w zasadzie dałaś to pole limit, żeby tylko jeden zwrócić

Reply:
Trzy batche myślę, że są spoko. Trzymanie wszystkich sprintów w pamięci też ma swoje wady :stuck_out_tongue:

Reply:
No tak ma :P
Możemy zrobić chwilowo tak, że zostawię to tak jak jest? A jak tylko wrócę z Erajdu (np w poniedziałek) to zrobię tak jak powinno być? bo teraz boję się to już rozgrzebywać, bo mam mniej więcej dwie godziny i będę musiał sie zbierać

Reply:
Albo nawet dwa batche: pierwszy z FINISHED, IN_PROGESS, PADDING, DECLARABLE, UPCOMING, a jak nie, to ostatni z `CLOSED` (w sensie chyba to jest tożsame z tym potrójnym, co napisałam)

Reply:
Spoks

Reply:
tak, to jest tożsame